iterator/src/main/java/com/iluwatar/iterator/list/App.java

```diff
@@ -30,7 +31,7 @@
  * The Iterator pattern is a design pattern in which an iterator is used to traverse a container and
  * access the container's elements. The Iterator pattern decouples algorithms from containers.
  * <p>
- * In this example the Iterator ({@link ItemIterator}) adds abstraction layer on top of a collection
+ * In this example the Iterator ({@link Iterator}) adds abstraction layer on top of a collection
```

Comment:
While investigating this change I found out that the current `TreasureChestIterator` implementation is incorrect. It is making assumptions about the underlying datastructure being `ArrayList`. And if it was changed to `LinkedList`, then iterator would perform really badly, which violates the whole purpose of iterator. So will have to investigate how to rectify that.